crates/tx5/src/backend/be_tx5_connection.rs

```diff
@@ -50,7 +52,22 @@ impl BackWaitCon for GoWaitCon {
                 _ => return Err(std::io::Error::other("already awaited")),
             };
 
-            con.ready().await;
+            tracing::info!(
+                "Waiting for connection to be ready for pub_key: {:?}",
+                pub_key
+            );
+            // This connection will only ready on a code path that successfully establishes either
+            // a WebRTC or a relayed connection. However, there are many exits from the code paths
+            // that attempt to set those up. In those cases, we own a semaphore permit that will
+            // never be released and we risk deadlocking here without this timeout.
+            tokio::time::timeout(timeout, con.ready())
```

Comment:
This ready check requires a semaphore to be dropped. This only happens if we successfully establish a network connection or fall back to the relay.

If something goes wrong here, we can end up waiting indefinitely for an event that won't happen. That is what leaves the connection stuck in a state where no messages sent.

If something does go wrong and we know about it, it'd be nice to notify this ready check immediately with an error... I will have a think about that on the coming PR that contains just the fix and not my debugging.